src/java/org/apache/cassandra/db/SSTableImporter.java

```diff
@@ -97,7 +99,19 @@ synchronized List<String> importNewSSTables(Options options)
                         try
                         {
                             abortIfDraining();
-                            verifySSTableForImport(descriptor, entry.getValue(), options.verifyTokens, options.verifySSTables, options.extendedVerify);
+
+                            if (options.verifySSTables || options.verifyTokens)
+                                verifySSTableForImport(descriptor, entry.getValue(), options.verifyTokens, options.verifySSTables, options.extendedVerify);
+
+                            if (options.failOnMissingIndex)
+                            {
+                                IndexDescriptor indexDescriptor = IndexDescriptor.create(descriptor,
+                                                                                         cfs.getPartitioner(),
```

Comment:
keep in mind that for SAI, the only partitioner currently supported is Murmur3Partitioner and it needs to be explicitly set on `CQLSSTableWriter` by `withPartitioner` (visible from tests in `CQLSSTableWriterTest`. 

Anyway, I still think that this is the correct way to do this. We should not hardcode `Murmur3Partitioner.instance` here even it is the only one which makes sense.